Demo

jotramon opened the pull request:
This is the first pitch, the app have all features of 1st pitch without connection with backend.
features:
- log in
- register
- add new team (with users)
- list teams
- raffle users of a team

jotramon committed:
primera versión demo

lfbeltran committed:
(all) add: color constants

lfbeltran committed:
(team) update: style and component

lfbeltran committed:
(teams) update: style

lfbeltran committed:
(all) add: eslint file

lfbeltran committed:
(all) install: react-native-star-rating

lfbeltran committed:
(feedback) add: screen

lfbeltran committed:
(team) update: style

lfbeltran committed:
(team) add: new team screen

lfbeltran committed:
(all) update: landing screen file

lfbeltran committed:
(team) update: team screen file

lfbeltran committed:
(login) add: view

lfbeltran committed:
Merge pull request #1 from platanus/demo-with-style

Demo with style

lfbeltran committed:
(login) add: view

lfbeltran committed:
(teams) update: add team view

lfbeltran committed:
(all) add: new packages

lfbeltran committed:
(team) add: new user to team view

lfbeltran committed:
(all) add: new package

lfbeltran committed:
(team) add: new user view

jotramon committed:
Merge pull request #2 from platanus/register-login

View login

lfbeltran committed:
(all) add: modal and emoji package

lfbeltran committed:
(feedback) fix: comma error

lfbeltran committed:
(team) add: choose user

lfbeltran committed:
(all) add: bottom-tabs

lfbeltran committed:
(all) update: tab views and hide header

lfbeltran committed:
(all) update: header

lfbeltran committed:
(team) update: size font title

cristobalilabaca committed:
added functionality for adding new team

jotramon committed:
incorporando agregar usuarios en agregar team

FelipeApablaza committed:
Add Splash, SignIn and SingUp Screens

jotramon committed:
merge

jotramon committed:
Merge branch 'newTeam' into users

jotramon committed:
Merge pull request #5 from platanus/register-login

Add Splash, SignIn and SingUp Screens

FelipeApablaza committed:
Merge branch 'demo' into sorteo

cristobalilabaca committed:
Changed map function with FlatList component

jotramon committed:
Merge pull request #4 from platanus/newTeam

added functionality for adding new team

jotramon committed:
Merge branch 'demo' into sorteo

jotramon committed:
Merge pull request #3 from platanus/sorteo

Sorteo

FelipeApablaza committed:
stack error fix

jotramon committed:
fixing errors

jotramon committed:
Merge branch 'users' into demo

jotramon committed:
Merge pull request #6 from platanus/demo

Demo

lfbeltran committed:
(team) update: style team view

lfbeltran committed:
(team) update: team view

jotramon committed:
Merge pull request #7 from platanus/style-small-fix

Style small fix

FelipeApablaza committed:
TeemScreen Components Update

jotramon committed:
Merge branch 'demo' into users

jotramon committed:
feat(addig-users): adding funtionality to add users in new teams

jotramon committed:
feat(addig-users): fixing

FelipeApablaza committed:
feat(adding users):fixing 2

jotramon committed:
feat(addig-users): fixing 3 and erreasing console.logs

jotramon committed:
feat(add-users): Pull requests changes and styling

cristobalilabaca committed:
Merge pull request #8 from platanus/users

Users

difernandez commented:
Hola! No alcancé a revisar el PR hoy por si acaso, el lunes tendré que verlo

difernandez commented on components/TeamScreen/Sorteo.js:
```diff
@@ -0,0 +1,54 @@
+
+
+
+const Sorteo = ({users, setVisible, visible, navigation}) => {
```
Traten de usar siempre inglés, tanto para nombre de constantes/funciones/variables como archivos/componentes

difernandez commented on components/TeamScreen/Sorteo.js:
```diff
@@ -0,0 +1,54 @@
+
+const Sorteo = ({users, setVisible, visible, navigation}) => {
+    const [userSelected, setUserSelected] = useState('');
+    console.log(userSelected)
```
Se pasó un `console.log`

difernandez commented on components/TeamScreen/Sorteo.js:
```diff
@@ -0,0 +1,54 @@
+                <Text style={styles.modalMessage}>El usuario seleccionado es:</Text>
+                <Text style={styles.modalUser}>{userSelected}</Text>
+                <Emoji name=":tada:" style={styles.modalEmoji} />
+                {/* <View style={styles.confirmButton}><Button title="¡Ayudanos con tu feedback!"  onPress={feedbackModal} /></View> */}
```
Está bien que esto quede comentado?

difernandez commented on components/TeamScreen/Sorteo.js:
```diff
@@ -0,0 +1,54 @@
+        setVisible(!visible);
+    };
+
+    const feedbackModal = () => {
```
el nombre de la función no se corresponde mucho con lo que hace. Si lo principal es que lleva a la screen de Feedback, algo como `goToFeedback` o `navigateToFeedback` sería más claro

difernandez commented on components/TeamScreen/TeamList.js:
```diff
@@ -0,0 +1,31 @@
+
+const User = (props) => {
+  return (
+    <TouchableWithoutFeedback onPress={() => {}}>
```
por qué usar `TouchableWithoutFeedback`? Por lo que veo en [los docs](https://reactnative.dev/docs/touchablewithoutfeedback) no se recomienda mucho usarlo, y además el `onPress` no hace nada

difernandez commented on components/TeamScreen/UsersList.js:
```diff
@@ -0,0 +1,32 @@
+
+const User = (props) => {
+  return (
+    <TouchableWithoutFeedback onPress={() => {}}>
```
Idem lo de `TouchableWithoutFeedback`

difernandez commented on components/TeamScreen/UsersList.js:
```diff
@@ -0,0 +1,32 @@
+
+
+
+export default UsersList
```
Creo que este y `TeamList` son iguale no? Por qué son dos componentes?

difernandez commented on screens/FeedbackScreen/FeedbackScreen.js:
```diff
@@ -0,0 +1,37 @@
+        
+        <View style={styles.startsContainer}>
+          <Text style={styles.title}>¡Ayúdanos a mejorar!</Text>
+          <StarRating disabled={false} maxStars={5} rating={stars} selectedStar={starsHanlder} fullStarColor={color.yellow}/>
```
`disabled` no es `false` por defecto?

difernandez commented on screens/LandingScreen/LandingScreen.js:
```diff
@@ -0,0 +1,67 @@
+
+const TeamView = (props) => {
+  return (
+    <TouchableOpacity onPress={() => props.navigation.navigate('Equipo', { name: props.team.item.name, description: props.team.item.description, members: props.team.item.members })}>
```
Cuando un callback no es tan simple podría mejorar la legibilidad extraerlo a una función

difernandez commented on screens/LandingScreen/LandingScreen.js:
```diff
@@ -0,0 +1,67 @@
+  )
+}
+
+// function LandingScreen({ navigation, route }) {
```
Se pasó este comentario

difernandez commented on screens/LandingScreen/LandingScreen.js:
```diff
@@ -0,0 +1,67 @@
+  };
+
+  useEffect(() => {
+    if (props.route.params?.name) {
+      const { name, description, members} = props.route.params
+      handlerTeam({
+        id: teamId, name, description, members: members, state: false,
+      });
+      setTeamId(teamId + 1);
+    }
```
Esto es por mientras que no se tiene conexión con el back, para que se vea un equipo creado no? Me imagino que eso después cambiaría por un post al back y que se guarde el nuevo team en algún lado

difernandez commented on screens/LoginScreen/LoginScreen.js:
```diff
@@ -0,0 +1,45 @@
+        </View>
+        <View style={styles.buttonContainer}>
+          <View style={styles.cancelButton}><Button title="Cancelar"  onPress={() => props.navigation.navigate('Equipos')} /></View>
+          <View style={styles.confirmButton}><Button title="Confirmar" onPress={() => navigation.navigate('Equipos', { name })} /></View>
```
- De dónde salió ese name?
- Como se le está dando un parámetro `name` no se activaría el `useEffect` del `LandingScreen`? está bien?

difernandez commented on screens/NewTeamScreen/NewTeamScreen.js:
```diff
@@ -0,0 +1,62 @@
+  }];
+  const {members} = props.route.params; 
+  const [name, setName] = React.useState('');
+  const [description, setDescription] = React.useState('');
```
Pueden importar el `useState` arriba para evitar repetir el `React.`

difernandez commented on screens/NewTeamScreen/NewTeamScreen.js:
```diff
@@ -0,0 +1,62 @@
+  const [name, setName] = React.useState('');
+  const [description, setDescription] = React.useState('');
+
+  const membersObjects = [];
+
+  if (members){
+    members.forEach(element => {
+      membersObjects.push(users.find(x => x.id === element));
+      
+    });
+  }
```
- No entiendo bien esto. Por qué hay members en las props y users en esta screen?
- Me entra la duda, esto no debería estar como valor inicial de un `useState` o dentro de un `useEffect`

difernandez commented on screens/NewUserToTeamScreen/NewUserToTeamScreen.js:
```diff
@@ -0,0 +1,82 @@
+  }, {
+    id: '4',
+    name: 'Javier Tramon',
+  }];
```
Como esto lo están usando en varios lados para simular lo que vedría del back podrían tenerlo en un archivo aparte

difernandez commented on screens/NewUserToTeamScreen/NewUserToTeamScreen.js:
```diff
@@ -0,0 +1,82 @@
+            // Estos items generan error, revisar
+            // itemFontFamily="system font"
+            // confirmFontFamily="system font"
+            // searchTextFontFamily="system font"
```
Se pasaron estas cosas comentadas parece

difernandez commented on screens/NewUserToTeamScreen/NewUserToTeamScreen.js:
```diff
@@ -0,0 +1,82 @@
+            // itemFontFamily="system font"
+            // confirmFontFamily="system font"
+            // searchTextFontFamily="system font"
+            selectText="Elige usuarios"
```
Esta está repetida

difernandez commented on screens/NewUserToTeamScreen/NewUserToTeamScreen.js:
```diff
@@ -0,0 +1,82 @@
+            // searchTextFontFamily="system font"
+            selectText="Elige usuarios"
+            searchInputPlaceholderText="Elige un usuario a agregar..."
+            onChangeInput={ (text)=> console.log(text)}
```
Se pasó este `console.log` parece

difernandez commented on screens/SignInScreen/SignInScreen.js:
```diff
@@ -0,0 +1,49 @@
+
+
+
+export default SignInScreen
```
Por qué hay un `SignInScreen` y un `LoginScreen`?

difernandez commented on screens/SignUpScreen/SignUpScreen.js:
```diff
@@ -0,0 +1,33 @@
+          </View>
+        </View>
+        <View style={styles.buttonContainer}>
+          <View style={styles.confirmButton}><Button title="Registrar"  /></View>
```
No debería llevar a "Equipos" o algo este botón?

difernandez commented on screens/TeamScreen/TeamScreen.js:
```diff
@@ -0,0 +1,65 @@
+  const [isModalVisible, setModalVisible] = useState(false);
+  const [userSelected, setUserSelected] = useState('');
+
+  const estoyEnEsteEquipo = true;
```
Idem lo de usar inglés

difernandez commented on screens/TeamScreen/TeamScreen.js:
```diff
@@ -0,0 +1,65 @@
+  } else {
+    const editButton = <Text />;
+    const sortButton = <Button title="Unirte a este grupo"  />;
+  }
```
Están redefiniendo constantes acá, y el valor inicial no se está usando ya que se pisa en el if/else. Si son valores que serán asignados después en este if/else, podrían ser inicialmente `let`s sin valor

difernandez commented on screens/TeamScreen/TeamScreen.js:
```diff
@@ -0,0 +1,65 @@
+          <Text style={styles.teamTitle}>{name}</Text>
+        </View>
+        <View style={styles.descriptionContainer}>
+          <Text style={styles.description}>{props.route.params.description}</Text>
```
la description se extrajo de `props.route.params`, se podría aprovechar de usar esa

difernandez commented on screens/TeamScreen/TeamScreen.js:
```diff
@@ -0,0 +1,65 @@
+        <Button title="Sortear" onPress={()=>setModalVisible(true)} />
+        {isModalVisible ?  (
+        <Sorteo visible={isModalVisible} setVisible={setModalVisible} users={members} navigation={props.navigation}/>
+        ) : <></> }
```
Si el `else` de la ternaria no renderea nada creo que podrían hacerlo con `&&`:
```
{isModalVisible &&
    <Sorteo visible={isModalVisible} setVisible={setModalVisible} users={members} navigation={props.navigation}/>
)}
```

difernandez reviewed (commented):
Les dejo algunos comentarios y dudas! Y aparte, un par de cosas un poco más generales:

- Traten de definir una convención de dónde quieren dejar los estilos. Esta un poco mezclada la cosa, con estilos en `styles.js`, o en una carpeta `styles` y en otro estaba en el mismo componente. Si no después se les pueden enredar las cosas
- Traten de no dejar tantas cosas comentadas. Se pueden agregar las cosas a medida que se necesiten, así queda más limpio el código y no van quedando comentarios olvidados que nadie quiere borrar por si acaso
- Traten de que estas revisiones sean más periódicas, con PRs para features más atómicas. Cuando se revisan muchas cosas es fácil que se pasen detalles y es más difícil aplicar cambios si se necesitan

Eso, ahí vean que cosas les hacen sentido! Y si tienen duda me cuentan por acá!

cristobalilabaca committed:
Changed variables languaje from spanish to english

cristobalilabaca committed:
Deleted all console.log

cristobalilabaca committed:
All styles files added to the styles folder

jotramon committed:
Fixing Code Review

Fixed:
- Duplicate components
- Sintaxis issues
- SingUp screen flux
- ternary operator fix
- erase commits
- fixing callback in landigscreen
- minor errors
